restapi/endpoints_test.go

```diff
@@ -0,0 +1,197 @@
+package restapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/suite"
+	"github.com/uwblueprint/shoe-project/internal/database/migrations"
+	"github.com/uwblueprint/shoe-project/internal/database/models"
+	"github.com/uwblueprint/shoe-project/testutils"
+	"gorm.io/gorm"
+)
+
+type endpointTestSuite struct {
+	suite.Suite
+	endpoint *httpexpect.Expect
+	db       *gorm.DB
+	token    string
+}
+
+func (suite *endpointTestSuite) SetupSuite() {
+	db, err := testutils.MockDatabase()
+	if err != nil {
+		suite.Fail("error while creating database", err)
+	}
+	suite.db = db
+
+	// setup required for jwt authentication
+	viper.SetDefault("auth.jwt_key", "random")
+	viper.SetDefault("auth.jwt_expiry", "5m")
+	viper.SetDefault("auth.jwt_issuer", "endpoint_tests")
+
+	router, err := Router(db)
+	if err != nil {
+		suite.FailNow("error while creating router", err)
+	}
+
+	server := httptest.NewServer(router)
+	suite.endpoint = httpexpect.New(suite.T(), server.URL)
+}
+
+func (suite *endpointTestSuite) SetupTest() {
+	if err := migrations.CreateTables(suite.db); err != nil {
+		suite.Fail("error while creating tables", err)
+	}
+	if err := migrations.CreateSuperUser(suite.db); err != nil {
+		suite.Fail("error creating super user", err)
+	}
+
+	suite.token = suite.endpoint.POST("/login").
```

Comment:
sorry for nit :/ 

i think this should be inserted directly into db too, instead of posting

Reply:
Don't we need a POST so we can grab the response?

Reply:
yes yes!!!

you're right, my bad.